app/worker/worker.go

```diff
@@ -81,145 +78,50 @@ func InitAndRun(
 		notifyPeriod = time.Second
 	}
 
-	channel := make(chan *natsio.Msg, concurrency)
+	jobChan := make(chan *natsio.Msg, concurrency)
 	unsubscribe, err := natsClient.JetStreamSubscribe(
 		ctx,
 		subject,
 		durableName,
-		channel,
+		jobChan,
 	)
 	if err != nil {
 		return errors.Wrap(err, "failed to subscribe to the nats JetStream")
 	}
-	defer func() {
-		_ = unsubscribe()
-	}()
 
 	quit := make(chan os.Signal, 1)
 	signal.Notify(quit, unix.SIGINT, unix.SIGTERM)
 
 	// Spawn worker pool
-	var wg = new(sync.WaitGroup)
-	wg.Add(concurrency)
+	wg := NewWorkGroup(jobChan, notifyPeriod, natsClient, dataStore)
 	for i := 0; i < concurrency; i++ {
-		workerID := i
-		go func() {
-			defer func() {
-				wg.Done()
-				cancel() // Unblock main thread if it's waiting
-			}()
-			wl := l.F(log.Ctx{"worker_id": workerID})
-			wCtx := log.WithContext(ctx, wl)
-
-			wl.Info("worker starting up")
-			workerMain(wCtx, channel, notifyPeriod, natsClient, dataStore)
-			wl.Info("worker shut down")
-		}()
+		go wg.RunWorker(ctx)
```

Comment:
nit: I would spawn the subroutine in the `RunWorker`, as it is now it is more like `WorkerMain`, besides that leaves open the modifications to the implementation details without changing the interface.

Reply:
I thought to use a similar naming convention as `os/exec`, where `Run` is blocking and `Start` (not implemented) forks and returns the process. I went for the `Run` interface since I didn't find much to put inside the wrapper.